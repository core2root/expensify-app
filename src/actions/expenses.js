import uuid from 'uuid';

//Add
export const addExpense = (
  { description = '', note = '', amount = 0, createdAt = 0 }
    = {}) => ({
      type: 'ADD_EXPENSE',
      expense: {
        id: uuid(),
        description: description,
        note: note,
        amount: amount,
        createdAt: createdAt
      }
    })

//Remove
export const removeExpense = ({ id } = {}) => ({
  type: 'REMOVE_EXPENSE',
  id: id
});

//Edit
export const editExpense = (id, updates) => ({
  type: 'EDIT_EXPENSE',
  id: id,
  updates: updates
});