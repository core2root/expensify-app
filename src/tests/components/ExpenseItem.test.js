import ExpenseItem from '../../components/expenseItem';
import {shallow} from 'enzyme';
import React from 'react';
import expenses from '../fixtures/expenses';

test('should render expense item with provided expense object', () => {
  const wrapper = shallow(<ExpenseItem  {...expenses[0]}/>);
  expect(wrapper).toMatchSnapshot();
});

