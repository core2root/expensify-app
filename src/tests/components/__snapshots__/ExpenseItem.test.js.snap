// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense item with provided expense object 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/1"
    >
      Gum
    </Link>
  </h3>
  <p>
    Amount: 
    195
     Created At: 
    Sun Dec 28 1969 02:00:00 GMT+0200 (Jerusalem Standard Time)
  </p>
</div>
`;
